### PR TITLE
3616 Run coverage tests with Python 3.6 on GitHub Actions

### PR DESCRIPTION
**Pull request by sajith**
Ticket is https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3616.  

We send coverage data to coveralls.io only from coverage tests running on GitHub Actions.  On GitHub Actions, we run coverage tests only with Python 2.7.  This PR will also run coverage tests using Python 3.6, thus giving us even more coverage reports.  It will also add `ubuntu-latest` to the matrix, because we can never have too little coverage data. ;-)

This PR supersedes PR #986, which has been split into two: #1021, which updated coveralls version for all jobs, and then this one.

### CODE REVIEW

**Comment by sajith** [comment]
I am not sure why coveralls bot has not left a comment on here yet.  I am also not sure why https://coveralls.io/builds/38441609 doesn't show coverage stats.  Perhaps they are related?

**Comment by sajith** [comment]
Maybe coveralls bot will leave a comment if this PR is switched to non-draft state?

**Comment by sajith** [comment]
Or is it because there's no change in coverage?

Going through https://coveralls.io/repos/32106/builds, it seems that older coveralls version (reports with longer IDs) reported minor amount of changes in coverage all the time, even though in reality there's no change in coverage.

Newer coveralls version (reports with short IDs) are more accurate than that.  No change in coverage, meaning nothing to report, meaning no comment posted here?

**Comment by sajith** [comment]
Coveralls made the comment, two hours after it received the last report, and now there are coverage numbers in https://coveralls.io/builds/38446290, https://coveralls.io/builds/38444035, and https://coveralls.io/builds/38441609.  

Ok, so it was not my fault. :-)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -20,8 +20,16 @@ jobs:
         os:
           - macos-latest
           - windows-latest
+          - ubuntu-latest
         python-version:
           - 2.7
+          - 3.6
+        exclude:
+          # Do not run coverage tests with Python 3.6 on Windows for
+          # now.  They will fail.  Dealing with them separately would
```
Comment (exarkun) [inline .github/workflows/ci.yml:29]: Sounds like a good thing for which to have a ticket?  Then this comment could link there and folks could discover more details (or fix it).

**Review by exarkun — APPROVED** [review]
Thanks.  One minor comment inline.  Please address and merge.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (sajith) [reply .github/workflows/ci.yml:29]: Filed https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3669 to track that, and updated the comment.

**Pull request merged**